### ISSUE TITLE
unit test coverage

### ISSUE THREAD

**Issue by hitman99** [movetokube/postgres-operator#32]
Code needs to be covered by unit tests, where possible, this issue is part of #31, which was split in half. Test coverage should be at least 80%. If some package is not testable, it must be refactored into a testable one.

**Comment by arnarg** [comment]
@hitman99 I've decided to go with [ginkgo](https://github.com/onsi/ginkgo) for writing unit tests, I hope you don't have any objections. You can see a tiny progress [here](https://github.com/movetokube/postgres-operator/commit/24e4f66b865b2bcfdf0dd7c71e4be4069efbb20b#diff-e21b623315ed09e70d71f18c78565e37).

**Comment by hitman99** [comment]
Not at all, I've used ginkgo in my past projects and it's fine. Thanks for hetring started with the tests!
